synchronicity/synchronizer.py

```diff
@@ -305,6 +306,7 @@ def _wrap_callable(self, f, interface, name=None, allow_futures=True):
                 )
             return f
 
+        @functools.wraps(f)
```

Comment:
This is enough to forward the type annotations from the underlying method I think. And it still seems to allow you to overwrite the name attribute on f_wrapped.